### PR TITLE
During call to connect, disconnect socket when create_session fails

### PR DESCRIPTION
**Pull request by harua8n**
Under certain conditions, normally when the server is under heavy load, it may fail creating session, if this happens we need to close the socket.

### CODE REVIEW

**Comment by oroulet** [comment]
Shouldn't we put all socket operations under try catch then? Just wondering...

**Comment by harua8n** [comment]
That might be the case but I can't test other scenarios though. If you want me I can do it.

**Comment by harua8n** [comment]
@oroulet, as discussed I have wrapped socket operations in try catch. Kindly review.

**Comment by harua8n** [comment]
Kindly review and let me know for inputs.

File: opcua/client/client.py [hunk 1]
```diff
@@ -206,36 +207,42 @@ def connect_and_get_server_endpoints(self):
...
         Connect, ask server for a list of known servers, and disconnect
         """
-        self.connect_socket()
-        self.send_hello()
-        self.open_secure_channel()  # spec says it should not be necessary to open channel
-        servers = self.find_servers()
-        self.close_secure_channel()
-        self.disconnect_socket()
+        try:
+            self.connect_socket()
```
Comment (oroulet) [inline opcua/client/client.py]: Not sure if that one should be before or after try: ..... Before i guess. If it fails then no socket is created and no need to close it

File: opcua/client/client.py [hunk 2]
```diff
@@ -246,7 +253,11 @@ def connect(self):
         self.connect_socket()
         self.send_hello()
```
Comment (oroulet) [inline opcua/client/client.py]: This one and the following line should be under try too. Isn't it?

**Comment by oroulet** [comment]
Sorry i missed it. Added some comments now

**Comment by harua8n** [comment]
Made changes as requested, kindly review again.

File: opcua/client/client.py [hunk 3]
```diff
@@ -6,6 +6,7 @@
 except ImportError:  # support for python2
     from urlparse import urlparse
 
+from concurrent.futures import _base
```
Comment (oroulet) [inline opcua/client/client.py]: why did you add that import? otherwise it looks ok

Comment (harua8n) [reply opcua/client/client.py]: In line 258 (call to self.create_session()), can sometimes fail and I need to clean-up socket connection if this happens.

Comment (harua8n) [reply opcua/client/client.py]: @oroulet, any additional thoughts? I can remove it but that means the user needs to know to clean-up on failure outside of the class.

Comment (harua8n) [reply opcua/client/client.py]: @oroulet, I have submitted new PR without importing concurrent.futures, kindly review.

**Comment by harua8n** [comment]
@oroulet, can you please review my latest changes?
